src/app/saga.test.ts

```diff
@@ -589,3 +601,72 @@ describe('appInit', () => {
     expect(initI18n).toHaveBeenCalledWith('en-US', true, '1')
   })
 })
+
+describe(requestInAppReview, () => {
+  const now = Date.now()
+  const oneDayAgo = now - ONE_DAY_IN_MILLIS
+  const fourMonthsAndADayAgo = now - ONE_DAY_IN_MILLIS * 121
+  beforeEach(() => {
+    jest.clearAllMocks()
+  })
+
+  it.each`
+    lastInteractionTimestamp | lastInteraction
+    ${null}                  | ${null}
+    ${fourMonthsAndADayAgo}  | ${'121 days ago'}
+  `(
+    `Should show when isAvailable: true, Last Interaction: $lastInteraction and Wallet Address: 0xTest`,
+    async ({ lastInteractionTimestamp }) => {
+      mocked(getFeatureGate).mockReturnValue(true)
+      mockIsInAppReviewAvailable.mockReturnValue(true)
+      await expectSaga(requestInAppReview, inAppReviewRequested(now))
+        .withState(
+          createMockStore({
+            web3: { account: '0xTest' },
+          }).getState()
+        )
+        .provide([[select(inAppReviewLastInteractionTimestampSelector), lastInteractionTimestamp]])
+        .run()
+
+      expect(InAppReview.RequestInAppReview).toHaveBeenCalledTimes(1)
```

Comment:
i may have commented in the other PR but i think it's important to check that the interaction timestamp is also updated here by checking that the correct action is dispatched. you can add a `.put()` before the `.run()` to do this :)

Reply:
I'll give this another shot today. I was trying something similar yesterday to bump up the test coverage.

Reply:
Took a trying to add a `.put` before the `.run` based on the examples of sagas which have tests and it's not working. I've taken a look at the [codecov](https://app.codecov.io/gh/valora-inc/wallet/blob/tomm%2Fin-app-review/src/app/saga.ts) for `src/app/saga.ts` and still can't get it to work.

Reply:
i created a PR to show how it can work https://github.com/valora-inc/wallet/pull/3729

there were 2 things that this test was missing:
- not locking the date - since the action relies on `Date.now`, we need this value to be predictable
- the `RequestInAppReview` function is an async one that returns a value. you were incorrectly mocking this as `jest.fn()` which returns nothing, so the action was never dispatched